docusaurus/docs/create-a-plugin/develop-a-plugin/best-practices.md

```diff
@@ -12,56 +12,69 @@ sidebar_position: 0.5
+- **Don't use the local file system** - Different plugins share the same environment. For security reasons, plugins shouldn't rely on local files.
+- **Don't use environment variables** - Environment variables are also a security risk and should be avoided. For configuration to a particular data source, use the `jsonData` or `secureJsonData` fields in the `plugin.json` file. If configuration is needed for the plugin which is shared among data sources, then use the [`plugin` configuration](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#pluginplugin_id).
+- **Plugins should not execute arbitrary code in the backend** - Again, this is a security risk and should be avoided. If your plugin needs to execute code, provide a list of allowed commands and validate the input before executing it.
+- **Don't expose sensitive information** - For security reasons, avoid exposing sensitive information such as secrets.
 - **In general, any error happening should be logged with level `error`.**
 - **Don't use the `info` level: use the `debug` level instead.**
 
 ## App plugins
 
-- **Specify a root page for your app.** If your app defines multiple pages, make sure to select a default one that will be used as a landing page for your plugin.
+- **Specify a root page for your app** - If your app defines multiple pages, make sure to select a default one that will be used as a landing page for your plugin.
 - **To generate dynamic apps, consider using [Grafana Scenes](https://grafana.com/developers/scenes/).**
-- **Consider  contributing a [UI extension](https://grafana.com/developers/plugin-tools/ui-extensions/)**. UI extensions can help a user to discover your app in context and continue a given workflow. Additionally, if your app provides context that be used in other apps, then create an extension point to allow these apps to do so, with no further changes required in your app.
+- **Consider contributing a [UI extension](https://grafana.com/developers/plugin-tools/ui-extensions/)** - UI extensions can help a user to discover your app in context and continue a given workflow. Additionally, if your app provides context that be used in other apps, then create an extension point to allow these apps to do so, with no further changes required in your app.
+
+## Publishing a plugin
+
+- **Add a GitHub badge** - Follow [these steps](https://grafana.com/blog/2021/01/21/6-tips-for-improving-your-grafana-plugin-before-you-publish/#tip-4-add-dynamic-badges-to-your-readme) to help users find your plugin using GitHub badges.
+- **Add workflow automation** - If your plugin is available on GitHub, be sure to [add the GitHub workflows](https://grafana.com/blog/2021/01/21/6-tips-for-improving-your-grafana-plugin-before-you-publish/#tip-5-automate-your-releases-using-github-actions) for plugin development to your repository.
```

Comment:
Nit: I would change this to "consider adding GitHub workflows" rather than "be sure", or something similar

```suggestion
- **Add workflow automation** - If your plugin is available on GitHub, consider [adding the GitHub workflows](https://grafana.com/blog/2021/01/21/6-tips-for-improving-your-grafana-plugin-before-you-publish/#tip-5-automate-your-releases-using-github-actions) for plugin development to your repository.
```

They're a nice addition, but not mandatory for plugins on GitHub. Some developers who are not familiar with GitHub actions may want to opt-out of those :smiley: